backend/fms_core/viewsets/sample.py

```diff
@@ -261,8 +264,16 @@ def update(self, request, *args, **kwargs):
 
     @action(detail=False, methods=["get"])
     def list_export(self, _request):
-        serializer = SampleExportSerializer(self.filter_queryset(self.get_queryset()), many=True)
-        return Response(serializer.data)
+        # Select related models in derived sample beforehand to improve performance and prefetch then in sample queryset
+        derived_samples = DerivedSample.objects.all().select_related('biosample', 'biosample__individual')
+
+        queryset = self.filter_queryset(self.get_queryset())
+        queryset = queryset.prefetch_related(Prefetch('derived_samples', queryset=derived_samples))
```

Comment:
I'm wondering if this would not work for just any sample queryset ? This could be inserted into the sample viewset get_queryset() by replacing self.get_queryset() by self.queryset ?

Reply:
I think it could work, we can try including it in get_queryset() function. It is a good suggestion.